payments-core/src/main/java/com/stripe/android/paymentsheet/model/SupportedPaymentMethod.kt

```diff
@@ -0,0 +1,36 @@
+package com.stripe.android.paymentsheet.model
+
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import com.stripe.android.R
+import com.stripe.android.model.PaymentMethod
+
+/**
+ * Enum defining all payment methods supported in Payment Sheet.
+ */
+internal enum class SupportedPaymentMethod(
+    val paymentMethodType: PaymentMethod.Type,
+    @StringRes val displayNameResource: Int,
+    @DrawableRes val iconResource: Int
+) {
+    Card(
+        PaymentMethod.Type.Card,
+        R.string.stripe_paymentsheet_payment_method_card,
+        R.drawable.stripe_ic_paymentsheet_add_pm_card
+    ),
+    Ideal(
+        PaymentMethod.Type.Ideal,
+        R.string.stripe_paymentsheet_payment_method_ideal,
+        R.drawable.stripe_ic_paymentsheet_add_pm_ideal
```

Comment:
I can imagine this being the thing that has the definition of the Form (see my review).  I would be nice if you could just create one of these and no other modifications to add a new payment method.